### PR TITLE
fix: add missing support for multiple entrypoints in same feed

### PR DESCRIPTION
**Pull request by digitalsadhu**
## JIRA Issue
[COREWEB-120](https://jira.finn.no/browse/COREWEB-120)
[COREWEB-181](https://jira.finn.no/browse/COREWEB-181)

## Status
**READY**

## Description
Changed to handle an array of feed items where `entrypoint: true` in reader instead of only ever returning a single feed item based on a return early approach.

## Todos
- [x] Tests

## Related PRs
* None

### CODE REVIEW

File: lib/feed-entrypoints.js [hunk 1]
```diff
@@ -7,7 +7,5 @@
  * @returns {object} - First entry in feed with property entry: true
  */
 module.exports = function feedEntrypoint(feed) {
-    for (const item of feed) {
-        if (item.entry) return item;
-    }
+    return feed.filter(item => Boolean(item.entry));
```
Comment (SimenB) [inline lib/feed-entrypoints.js]: is `Boolean` needed? it should either be a boolean already or `undefined`

File: lib/reader.js [hunk 2]
```diff
@@ -39,8 +39,10 @@ module.exports = async (feeds, options) => {
     await Promise.all(
         feeds.map(async feed => {
             const root = join(opts.directory, await hashFeed(feed));
-            const entrypoint = feedEntrypoint(feed);
-            entries.push(join(root, entrypoint.file));
+            const entrypoints = feedEntrypoints(feed);
```
Comment (SimenB) [inline lib/reader.js]: I'd do 
```js
feedEntrypoints(feed).forEach(entrypoint => entries.push(join(root, entrypoint.file)));
```

but doesn't really matter.

Comment (digitalsadhu) [reply lib/reader.js]: Cool, yea could do. Think i'll just plow ahead for now though.

File: lib/feed-entrypoints.js (hunk 1 above)
Comment (digitalsadhu) [reply lib/feed-entrypoints.js]: yea true. Plowing ahead for now... :plow <-- really no plow emoji?

File: lib/reader.js (hunk 2 above)
Comment (SimenB) [reply lib/reader.js]: or even just inline the `feed-entrypoints.js` file and do
```js
feed
  .filter(item => Boolean(item.entry))
  .map(entrypoint => join(root, entrypoint.file))
  .forEach(file => entries.push(file));
```

File: lib/feed-entrypoints.js (hunk 1 above)
Comment (SimenB) [reply lib/feed-entrypoints.js]: there really should be a plow emoji

**Pull request merged**